docs/_guide/events.md

```diff
@@ -8,96 +8,156 @@ slug: events
+  await new Promise((r) => setTimeout(r, 0));
+  this.addEventListener('click', this._handleClick);
+}
+```
+
+`firstUpdated` fires after the first time your component has been updated and called its `render` method, but **before the browser has had a chance to paint**. The `Promise`/`setTimeout` line yields to the browser
     
-    ```js
-    connectedCallback() {
-      super.connectedCallback();
-      document.addEventListener('readystatechange', this.handleChange);
-    }
-    disconnectedCallback() {
-      document.removeEventListener('readystatechange', this.handleChange);
-      super.disconnectedCallback();
-    }
-    ```
+See [firstUpdated](/guide/lifecycle#firstupdated) in the Lifecycle documentation for more information.
+
 
-### Using `this` in event handlers
+## Using `this` in event listeners
 
-The default JavaScript context object (`this`) inside an event handler belonging to a LitElement-based component is the component itself. 
+Event listeners added using the declarative (`@event`) syntax in the template are automatically _bound_ to the component.
 
-Therefore, you can use `this` to refer to your element instance inside any event handler:
+Therefore, you can use `this` to refer to your component instance inside any declarative event handler:
 
 ```js
 class MyElement extends LitElement {
   render() {
-    return html`<button @click="${this.handleClick}">click</button>`;
+    return html`<button @click="${this._handleClick}">click</button>`;
   }
-  handleClick(e) {
+  _handleClick(e) {
     console.log(this.prop);
   }
 }
 ```
 
+When adding listeners imperatively with `addEventListener`, you'll need to bind the event listener yourself if you need a reference to the component instance. For example:
+
+```js
+this.boundResizeHandler = this.handleResize.bind(this);
+window.addEventListener('resize', this.boundResizeHandler);
+```
+
 See the [documentation for `this` on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) for more information.
 
-### Use cases
+## Setting event listener options
+
+When you add an event listener imperatively, using `addEventListener`, you can specify various event listener options. For example, to use a capture phase event listener in plain JavaScript you'd do something like this:
+
+```js
+someElement.addEventListener('click', this._handleClick, {capture: true});
+```
+
+The `eventOptions` decorator allows you to add event listener options to a listener that's added declaratively in your template.
+
+```js
+import {LitElement, html, eventOptions} from 'lit-element';
+...
+
+@eventOptions({capture: true})
+_handleClick_() { ... }
+
+render() { 
+  return html`
+    <button @click=${this._handleClick}>Click me!</button>
+  `;
+}
+```
+
+<div class="alert alert-info">
+
+**Using decorators.** Decorators are a proposed JavaScript feature, so you’ll need to use a compiler like Babel or the TypeScript compiler to use decorators. See Using decorators for details.
```

Comment:
maybe?

```suggestion
**Using decorators.** Decorators are a proposed JavaScript feature, so you’ll need to use a compiler like Babel or TypeScript to use decorators. See Using decorators for details.
```

Reply:
Updated this in all the places it appears.